bigquery/query.go

```diff
@@ -374,19 +377,31 @@ func (q *Query) Read(ctx context.Context) (it *RowIterator, err error) {
 	ctx = trace.StartSpan(ctx, "cloud.google.com/go/bigquery.Query.Run")
 	defer func() { trace.EndSpan(ctx, err) }()
 	queryRequest, err := q.probeFastPath()
-	if err != nil {
+	if err != nil || q.forceStorageAPI {
 		// Any error means we fallback to the older mechanism.
 		job, err := q.Run(ctx)
 		if err != nil {
 			return nil, err
 		}
 		return job.Read(ctx)
 	}
+
 	// we have a config, run on fastPath.
 	resp, err := q.client.runQuery(ctx, queryRequest)
 	if err != nil {
 		return nil, err
 	}
+
+	// If more pages are available, use the Storage API instead
+	if resp.PageToken != "" && q.client.rc != nil {
```

Comment:
This looks incorrect.    As written, this appears to force the user to re-run the job a second time if its a multi page result and complete.  inside the jobComplete check, we should resolve which iterator to use.

I don't think we get dest table in the query response, so we may need to pay the cost of a jobs.get call here.